src/XMakeBuildEngine/Evaluation/ProjectRootElementCache.cs

```diff
@@ -236,12 +236,16 @@ internal ProjectRootElementCache(bool autoReloadFromDisk)
                             // the cache as we get test coverage of the rest of the cache code.
                             XmlDocument document = new XmlDocument();
                             document.PreserveWhitespace = projectRootElement.XmlDocument.PreserveWhitespace;
-							
-                            XmlReaderSettings xmlReaderSettings = new XmlReaderSettings();
-                            xmlReaderSettings.DtdProcessing = DtdProcessing.Ignore;
-                            using (XmlReader xr = XmlReader.Create(File.OpenRead(projectRootElement.FullPath), xmlReaderSettings))
+
+                            XmlReaderSettings dtdSettings = new XmlReaderSettings();
+                            dtdSettings.DtdProcessing = DtdProcessing.Ignore;
+
+                            // Note 4096 and FileOptions.SequentialScan are set by the legacy constructor of StreamReader(string path) which is
+                            // not available in .NET Core.
+                            using (var stream = new FileStream(projectRootElement.FullPath, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, FileOptions.SequentialScan))
+                            using (XmlReader xtr = XmlReader.Create(stream, dtdSettings))
```

Comment:
I followed the old `XmlTextReader` in the debugger. It [calls](https://github.com/dotnet/corefx/blob/62e4a0ca7911b2e1d823fb3461ebf2a26fa62150/src/System.Private.Xml/src/System/Xml/XmlDownloadManager.cs#L28)

```C#
FileStream(String path, FileMode mode, FileAccess access, FileShare share, int bufferSize)
```

with 

```C#
return new FileStream( uri.LocalPath, FileMode.Open, FileAccess.Read, FileShare.Read, 1 );
```

Since having an explicit `StreamReader` is new, I'm not sure it's worth jumping through hoops to preserve the default. Can we not just use

```C#
new FileStream(projectRootElement.FullPath, FileMode.Open, FileAccess.Read, FileShare.Read)
```

?